lib/opencensus/trace/integrations/faraday_middleware.rb

```diff
@@ -17,53 +17,87 @@
 module OpenCensus
   module Trace
     module Integrations
+      ##
+      # A middleware for Faraday that creates spans for outgoing requests, and
+      # propagates the trace context appropriately.
+      #
       class FaradayMiddleware < Faraday::Middleware
-        ##
-        # Create a Trace span with the HTTP request/response information.
-        def call env
-          OpenCensus::Trace.in_span "faraday_request" do |span|
-            add_request_labels span, env if span
-
-            response = @app.call env
+        ## The default name for Faraday spans
+        DEFAULT_SPAN_NAME = "faraday_request".freeze
 
-            add_response_labels span, env if span
+        ##
+        # Create a FaradayMiddleware.
+        #
+        # @param [#call] app Next item on the middleware stack.
+        # @param [SpanContext] span_context The span context within which
+        #     to create spans. Optional: If omitted, spans are created in the
+        #     current thread-local span context.
+        # @param [String, #call] span_name The name of the span to create.
+        #     Can be a string or a callable that takes a faraday request env
+        #     and returns a string. Optional: If omitted, uses
+        #     `DEFAULT_SPAN_NAME`
+        # @param [#call] sampler The sampler to use when creating spans.
+        #     Optional: If omitted, uses the sampler in the current config.
+        #
+        def initialize app, span_context: nil, span_name: nil, sampler: nil
+          @app = app
+          @span_context = span_context || OpenCensus::Trace
+          @span_name = span_name || DEFAULT_SPAN_NAME
+          @sampler = sampler
+        end
 
-            response
+        ##
+        # Wraps an HTTP call with a span with the request/response info.
```

Comment:
This one is missing the YARD param definition or should be marked as internal.

Reply:
Fixed